web/static/js/components/simulation/SimulationSteps.jsx

```diff
@@ -0,0 +1,153 @@
+// @flow
+import React, { Component } from 'react'
+import { icon } from '../../step'
+import { translate } from 'react-i18next'
+import { Card, UntitledIfEmpty } from '../ui'
+import classNames from 'classnames/bind'
+import { flatMapDepth } from 'lodash'
+import propsAreEqual from '../../propsAreEqual'
+
+type Step = {
+  type: string,
+  title: string,
+  id: string,
+  steps: ?Array<Step>,
+  status: string,
+  response: ?string
+}
+
+type Submission = {
+  stepId: string,
+  response: ?string
+}
+
+type Props = {
+  submissions: Array<Submission>,
+  steps: Array<Step>,
+  currentStepId: string,
+  t: Function,
+  simulationIsEnded: boolean
+}
+
+type State = {
+  headedBySectionSteps: Array<Step>
+}
+
+class SimulationSteps extends Component<Props, State> {
+  constructor(props) {
+    super(props)
+    this.state = this.stateFromProps(props)
+  }
+
+  componentWillReceiveProps(newProps) {
+    if (propsAreEqual(this.props, newProps)) return
+    this.setState(this.stateFromProps(newProps))
+  }
+
+  stateFromProps(props) {
+    const { steps, submissions, simulationIsEnded, currentStepId } = props
+
+    const isActive = (step: Step): boolean => {
+      return step.type == 'section'
+        ? !!step.steps && step.steps.some(s => s.id == currentStepId)
+        : currentStepId == step.id
+    }
+
+    const isCompleted = (step: Step, isActive: boolean): boolean => {
+      if (isActive) return false
+      return step.type == 'section'
+        ? !!step.steps && step.steps.some(st => submissions.some(su => su.stepId == st.id))
+        : submissions.some(su => su.stepId == step.id)
+    }
+
+    const wasSkipped = (step: Step, isActive: boolean, isCompleted: boolean, headedBySectionSteps: Array<Step>): boolean => {
+      if (isActive || isCompleted) return false
+      if (simulationIsEnded) {
+        // if the simulation is ended (and the step is neither active nor completed)
+        // then, it was skipped
```

Comment:
Append comment that section will be marked as completed because isCompleted will return true.

Otherwise it seems that section will be marked as skipped with simulation ended.

Reply:
I modified how this is expressed in [this commit](https://github.com/instedd/surveda/pull/1698/commits/75372fcd8aacf8f74973733085179c0ac2b73b98), in the context of the `stepStatus` extraction.